HW2/P3/P3.txt

```diff
@@ -0,0 +1,55 @@
+4 Threads: 1074.94829 Million Complex FMAs in 0.582358121872 seconds, 1845.85438002 million Complex FMAs / second
+4 Threads: 1074.94829 Million Complex FMAs in 0.503425121307 seconds, 2135.26946611 million Complex FMAs / second
+4 Threads: 1074.94829 Million Complex FMAs in 0.496190071106 seconds, 2166.40427247 million Complex FMAs / second
+4 Threads: 1074.94829 Million Complex FMAs in 0.510525941849 seconds, 2105.57035771 million Complex FMAs / second
+4 Threads: 1074.94829 Million Complex FMAs in 0.514708995819 seconds, 2088.4583303 million Complex FMAs / second
+4 Threads: 1074.94829 Million Complex FMAs in 0.65876698494 seconds, 1631.75798814 million Complex FMAs / second
+
+
+----------------------------------------------
+Without Instruction Level Parallelism (this was done on original code before modifying it for instruction-level parallelism)
+See without_instruction_parallelism.png for a graph of this.
+
+
+1 Threads: 1074.656613 Million Complex FMAs in 6.1589179039 seconds, 174.487893777 million Complex FMAs / second
+1 Threads: 1074.656613 Million Complex FMAs in 6.2493751049 seconds, 171.962251419 million Complex FMAs / second
+1 Threads: 1074.656613 Million Complex FMAs in 6.19680500031 seconds, 173.421079564 million Complex FMAs / second
+1 Threads: 1074.656613 Million Complex FMAs in 6.42867398262 seconds, 167.16613969 million Complex FMAs / second
+1 Threads: 1074.656613 Million Complex FMAs in 6.1917579174 seconds, 173.562440156 million Complex FMAs / second
+1 Threads: 1074.656613 Million Complex FMAs in 6.31927990913 seconds, 170.05997969 million Complex FMAs / second
+
+2 Threads: 1074.656613 Million Complex FMAs in 3.37078309059 seconds, 318.815119252 million Complex FMAs / second
+2 Threads: 1074.656613 Million Complex FMAs in 3.36708903313 seconds, 319.1648936 million Complex FMAs / second
+2 Threads: 1074.656613 Million Complex FMAs in 3.39102196693 seconds, 316.912312418 million Complex FMAs / second
+2 Threads: 1074.656613 Million Complex FMAs in 3.36820602417 seconds, 319.059049621 million Complex FMAs / second
+2 Threads: 1074.656613 Million Complex FMAs in 3.37363100052 seconds, 318.545985864 million Complex FMAs / second
+2 Threads: 1074.656613 Million Complex FMAs in 3.36886405945 seconds, 318.996728285 million Complex FMAs / second
+
+4 Threads: 1074.656613 Million Complex FMAs in 1.85349988937 seconds, 579.798584916 million Complex FMAs / second
+4 Threads: 1074.656613 Million Complex FMAs in 1.86611509323 seconds, 575.879063889 million Complex FMAs / second
+4 Threads: 1074.656613 Million Complex FMAs in 1.84833502769 seconds, 581.418734644 million Complex FMAs / second
+4 Threads: 1074.656613 Million Complex FMAs in 2.00391697884 seconds, 536.278011689 million Complex FMAs / second
+4 Threads: 1074.656613 Million Complex FMAs in 1.88849616051 seconds, 569.054168853 million Complex FMAs / second
+4 Threads: 1074.656613 Million Complex FMAs in 2.02837705612 seconds, 529.811067305 million Complex FMAs / second
+
```

Comment:
maybe you could comment a little more about it!
But the results seem very good.